src/packet/public_key_encrypted_session_key.js

```diff
@@ -125,6 +125,9 @@ class PublicKeyEncryptedSessionKeyPacket {
     if (this.publicKeyAlgorithm !== key.algorithm) {
       throw new Error('Decryption error');
     }
+    if (key.isStoredInHardware() && !plugin){
```

Comment:
Can we pass `config`, rather than `plugin`, all the way here as well?

Reply:
Done! Config should be now passed where possible. In a couple of places I've left `plugin_with_data` object to keep state for the concurrent keys generation.